### PR TITLE
(#669) Add version completion for install and upgrade commands

### PR DESCRIPTION
**Pull request by AdmiringWorm**
## Description Of Changes

Enhances PowerShell tab completion for Chocolatey `install` and `upgrade` commands by dynamically listing available `--version` values for a specified package. Specifically:

- Adds a new `chocoRemotePackageVersions` helper function to query all available versions for a given package via `choco search`.
- Updates `ChocolateyTabExpansion` to detect version requests and emit completions using a match on partial version input.
- Improves Pester coverage by adding various test cases that validate behavior for:
  - Empty, quoted, and partially typed `--version` values.
  - Filtering of results based on major version input (e.g., `2` returns only 2.x).

## Motivation and Context

Users installing or upgrading packages via Chocolatey sometimes need to specify a version manually. Without completions, this requires knowing exact version strings in advance. This change improves usability by providing context-aware completions for the `--version` parameter based on available versions from the remote source. It also adds comprehensive tests to verify correctness and edge case behavior.

## Testing

1. Manually tested interactive tab completion for `choco install` and `choco upgrade` using various `--version` inputs. (`--version=`, `--version='` and `--version=''`. Latest need to have the cursor between the single quotes).
2. Added Pester tests to verify (_Not yet run_):
   - Completions return the correct version set.
   - Filtering is applied based on user-typed prefix.
   - Behavior differs when the cursor is placed before vs. after closing quotes.

### Operating Systems Testing

- Windows 11

## Change Types Made

* [ ] Bug fix (non-breaking change).
* [x] Feature / Enhancement (non-breaking change).
* [ ] Breaking change (fix or feature that could cause existing functionality to change).
* [ ] Documentation changes.
* [x] PowerShell code changes.

## Change Checklist

* [ ] Requires a change to the documentation.
* [ ] Documentation has been updated.
* [x] Tests to cover my changes, have been added.
* [ ] All new and existing tests passed? (_Not run yet, will run as part of this PR_)
* [x] PowerShell code changes: PowerShell v3 compatibility checked?

## Related Issue

- Related to #669 
- PROJ-1501

### CODE REVIEW

File: src/chocolatey.resources/helpers/ChocolateyTabExpansion.ps1 [hunk 1]
```diff
@@ -160,6 +160,17 @@ function script:chocoRemotePackages($filter) {
         ForEach-Object { $_.Split('|')[0] }
 }
 
+function script:chocoRemotePackageVersions($Name, $Version) {
+        $packageVersions = & $script:choco search $Name --exact --all-versions --limit-output |
```
Comment (corbob) [inline src/chocolatey.resources/helpers/ChocolateyTabExpansion.ps1]: Not sure if this will extend to other parts of tab completion beyond this PR, but since we now have `--include-headers`, we should be able to use it here.

Comment (AdmiringWorm) [reply src/chocolatey.resources/helpers/ChocolateyTabExpansion.ps1]: That is fair, I didn't think of using it.

I think this is the only place where we do the easy way out to convert it from CSV.

I'll look at updating this method.

Comment (AdmiringWorm) [reply src/chocolatey.resources/helpers/ChocolateyTabExpansion.ps1]: I have updated this call to use the `--include-headers` flag.

**Comment by gep13** [comment]
@AdmiringWorm is this still in draft, or is this ready for review?

**Comment by AdmiringWorm** [comment]
> @AdmiringWorm is this still in draft, or is this ready for review?

It is still in draft, waiting on the test kitchen to finish.
Once it have finished and succeed I will get this PR rebased and open it up for review.

**Comment by AdmiringWorm** [comment]
@gep13 I have opened this for review now, it succeeded internally on the 2016 image, but failed to spin up the 2019 image (but since it succeeded on one of them, I think it is fine).

**Review by gep13 — APPROVED** [review]
LGTM!

**Comment by gep13** [comment]
@AdmiringWorm this looks great! Thank you for taking the time to get this added!  This will be a big help going forward!
